core/src/main/java/ai/timefold/solver/core/impl/domain/variable/descriptor/VariableDescriptor.java

```diff
@@ -139,4 +140,22 @@ public final boolean isGenuineAndUninitialized(Object entity) {
                 .variable(variableName);
         return cachedMetamodel;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o instanceof VariableDescriptor<?> other) {
+            // VariableDescriptors should be compared by identity, but
+            // it appears two different instances of the same VariableDescriptor
+            // get created somewhere
+            // VariableDescriptors can be compared by their entity classes and names
+            return Objects.equals(entityDescriptor.getEntityClass(), other.entityDescriptor.getEntityClass())
+                    && Objects.equals(variableName, other.variableName);
+        }
+        return false;
```

Comment:
Let's not put a band-aid on a problem. Let's find where it happens, and fix it at the source please.